File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
  const stack = [];

  for (let i = 0; i < s.length; i++) {
    if (stack[stack.length - 1] === '(' && s[i] === ')') {
      stack.pop();
    } else {
      stack.push(s[i]);
    }
  }

  // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
  // console.log('Hello Javascript')

  return !stack.length
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    
    for s in s:
        if s == '(':
            stack.append(s)
        else:
            if stack:
                stack.pop()
            else:
                answer = False
                break
                    
    if stack:
        answer = False
    
    return answer